crates/sui-indexer-alt/migrations/2024-10-21-003426_tx_indices/up.sql

```diff
@@ -0,0 +1,39 @@
+CREATE TABLE IF NOT EXISTS tx_affected_addresses (
+    tx_sequence_number          BIGINT       NOT NULL,
+    affected                    BYTEA        NOT NULL,
+    sender                      BYTEA        NOT NULL,
+    PRIMARY KEY(affected, tx_sequence_number)
+);
+
+CREATE INDEX IF NOT EXISTS tx_affected_addresses_tx_sequence_number_index ON tx_affected_addresses (tx_sequence_number);
+CREATE INDEX IF NOT EXISTS tx_affected_addresses_sender ON tx_affected_addresses (sender, affected, tx_sequence_number);
+
+CREATE TABLE IF NOT EXISTS tx_digests (
+    tx_digest                   BYTEA        PRIMARY KEY,
+    tx_sequence_number          BIGINT       NOT NULL
+);
+CREATE INDEX IF NOT EXISTS tx_digests_tx_sequence_number_index ON tx_digests (tx_sequence_number);
```

Comment:
> I just realised something today -- because kv_transactions is now keyed by tx_digest this table should actually go in the opposite direction: It should be keyed by tx_sequence_number so that we can map sequence numbers to digests.

This part makes sense to me. When the user input is a list of transaction digests, we look them up directly in `kv_transactions`. And when user queries transactions via optional filters, we first look up the transaction sequence numbers using transaction indices, then use `tx_digests` to map the sequence numbers to the corresponding digests, and then finally look them up in `kv_transactions` via the digests. So that's three tables touched. 

Writing this out, I see why you are thinking if we can use digest everywhere as this will allow us to remove one layer of indirection. But even if we use transaction digest everywhere, we will still need to store transaction sequence number everywhere as well for cursor and scan limit. That is one extra column (digest) for each row in five tables (affected objects + addresses, tx kinds, balance changes, tx calls fun), not to mention the indices we need to add here.

Reply:
I suppose we can punt on this discussion right now and merge this PR in. I've created a linear task for it.

Reply:
Ah yes good point on why we still want the sequence number. Indeed, happy to punt on whether we need to add `tx_digest` everywhere (and get rid of the explicit `tx_digests` table).